pkg/cli/flags_test.go

```diff
@@ -1,56 +1,58 @@
 package cli
 
 import (
-	"context"
-	"flag"
+	"bytes"
+	"fmt"
+	"os"
 	"time"
 
 	. "github.com/onsi/ginkgo"
 	. "github.com/onsi/gomega"
-	"github.com/peterbourgon/ff/v3"
-	"github.com/peterbourgon/ff/v3/ffcli"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
 
 	"github.com/pyroscope-io/pyroscope/pkg/config"
 	"github.com/pyroscope-io/pyroscope/pkg/util/bytesize"
 )
 
 type FlagsStruct struct {
-	Config   string
-	Foo      string
-	Foos     []string
-	Bar      int
-	Baz      time.Duration
-	FooBar   string
-	FooFoo   float64
-	FooBytes bytesize.ByteSize
+	Config   string            `mapstructure:"config"`
+	Foo      string            `mapstructure:"foo"`
+	Foos     []string          `mapstructure:"foos"`
+	Bar      int               `mapstructure:"bar"`
+	Baz      time.Duration     `mapstructure:"baz"`
+	FooBar   string            `mapstructure:"foo-bar"`
+	FooFoo   float64           `mapstructure:"foo-foo"`
+	FooBytes bytesize.ByteSize `mapstructure:"foo-bytes"`
 }
 
 var _ = Describe("flags", func() {
 	Context("PopulateFlagSet", func() {
 		Context("without config file", func() {
 			It("correctly sets all types of arguments", func() {
-				exampleFlagSet := flag.NewFlagSet("example flag set", flag.ExitOnError)
-				cfg := FlagsStruct{}
-				PopulateFlagSet(&cfg, exampleFlagSet)
-
-				exampleCommand := &ffcli.Command{
-					FlagSet: exampleFlagSet,
-					Exec: func(_ context.Context, args []string) error {
+				exampleCommand := &cobra.Command{
+					RunE: func(cmd *cobra.Command, args []string) error {
 						return nil
 					},
 				}
 
-				err := exampleCommand.ParseAndRun(context.Background(), []string{
-					"-foo", "test-val-1",
-					"-foos", "test-val-2",
-					"-foos", "test-val-3",
-					"-bar", "123",
-					"-baz", "10h",
-					"-foo-bar", "test-val-4",
-					"-foo-foo", "10.23",
-					"-foo-bytes", "100MB",
+				cfg := FlagsStruct{}
+				PopulateFlagSet(&cfg, exampleCommand.Flags())
+
+				b := bytes.NewBufferString("")
+				exampleCommand.SetOut(b)
+				exampleCommand.SetArgs([]string{
+					fmt.Sprintf("--foo=%s", "test-val-1"),
```

Comment:
My biggest concern is backward compatibility. Is there a way to accept [single-dash args](https://github.com/spf13/cobra/issues/740)?